### PR TITLE
Log system cleanup

### PR DESCRIPTION
**Pull request by hrydgard**
For ancient reasons, we've been stuck with two logging systems.

This removes the base/logging.h one, and replaces all uses with calls into the Common/Log.h one. 

I plan to later add a nice in-app log viewer on Android (for easier reporting from users) and for that, all logging needs to be in the same system.

### CODE REVIEW

**Comment by hrydgard** [comment]
Squished away a lot of commits, this got too messy. But now it's done.

File: Common/LogManager.cpp [hunk 1]
```diff
@@ -93,7 +93,9 @@ static const LogNameTableEntry logTable[] = {
 	{LogTypes::MEMMAP,     "MEMMAP"},
 	{LogTypes::SASMIX,     "SASMIX"},
 	{LogTypes::SAVESTATE,  "SAVESTATE"},
-	{LogTypes::FRAMEBUF,   "FRAMEBUF"},
+	{LogTypes::FRAMEBUF,  "FRAMEBUF"},
```
Comment (unknownbrackets) [inline Common/LogManager.cpp:101]: nit: Alignment broken?

-[Unknown]

File: Core/TextureReplacer.cpp [hunk 2]
```diff
@@ -380,15 +380,11 @@ void TextureReplacer::PopulateReplacement(ReplacedTexture *result, u64 cachekey,
 static bool WriteTextureToPNG(png_imagep image, const std::string &filename, int convert_to_8bit, const void *buffer, png_int_32 row_stride, const void *colormap) {
 	FILE *fp = File::OpenCFile(filename, "wb");
 	if (!fp) {
-		ERROR_LOG(SYSTEM, "Unable to open texture file for writing.");
+		ERROR_LOG(IO, "Unable to open texture file for writing.");
 		return false;
 	}
 
 	if (png_image_write_to_stdio(image, fp, convert_to_8bit, buffer, row_stride, colormap)) {
-		if (fclose(fp) != 0) {
```
Comment (unknownbrackets) [inline Core/TextureReplacer.cpp:388]: Hmm, we still want to fclose though no?  Otherwise it's a leak.

-[Unknown]

File: UI/NativeApp.cpp [hunk 3]
```diff
@@ -1406,9 +1408,6 @@ void NativeShutdown() {
 
 	g_Discord.Shutdown();
 
-	delete logger;
```
Comment (unknownbrackets) [inline UI/NativeApp.cpp:1409]: Hm, but we still new in init.  Should we check if `!logger` there?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]
I'm a big fan.

-[Unknown]

File: UI/NativeApp.cpp (hunk 3 above)
Comment (hrydgard) [reply UI/NativeApp.cpp:1409]: Hm I removed this intending to change how LogManager deals with their lifetimes but then backed out and forgot this. Nice catch.

File: Core/TextureReplacer.cpp (hunk 2 above)
Comment (hrydgard) [reply Core/TextureReplacer.cpp:388]: Oops, went a little too fast here...

**Comment by hrydgard** [comment]
Green on Travis - I'm gonna merge. Hopefully nothing has broken :)

**Pull request merged**